torchtitan/experiments/deepseek_v3/model.py

```diff
@@ -520,6 +513,41 @@ def __init__(self, config):
                 config=config, intermediate_size=intermediate_size
             )
 
+        # Group Gemm
+        self._initialize_group_gemm_strategies()
+
+        assert (
+            self.group_mm in self.group_gemm_strategies
+        ), f"selected group gemm {self.group_mm} is not avaiable!"
+        # keep active gg ready
+        self.group_gemm_instance = self.group_gemm_strategies[self.group_mm]
+        self._buffer_initialized = False
+
+    def _initialize_group_gemm_strategies(self):
+        """Initialize available group GEMM strategies"""
+        self.group_gemm_strategies = {
```

Comment:
nit: can you make `group_gemm_strategies` a global variable (or a class variable) so that we don't need to create it per MoE instance? 

Also, it seems we can create the strategy instances unconditionally regardless of whether they are available.
```
group_gemm_strategies = {
    "torch": TorchBF16GroupGEMM(MLP.act_fn),
    ""torchao": TorchAOBF16GroupGEMM(MLP.act_fn),
    ...
}
```

Reply:
moved it to be a class variable to cut the overhead. 
re: make strategy instances regardless - I don't think that's a good idea as by only creating the valid instsances we can quickly abort with a reasonable message if the user selects something not supported/available (vs. crashing out with a random error)?